modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/documentation/DocumentationHelper.java

```diff
@@ -134,15 +148,54 @@ public static String getHtml(String doc) {
      * Makes sure no passwords are embedded in the XML.
      */
     public static String secureXML(String xml) {
-        if (xml == null || !xml.contains("assword")) {
+        if (StringUtils.isBlank(xml)) {
             return xml;
         }
-        xml = xml.replaceAll("<([a-zA-Z]*[pP])assword>[^<]*</([a-zA-Z]*)assword>", "<$1assword>********</$2assword>");
-        // attributes: nuxeo-core-auth
-        xml = xml.replaceAll("([a-zA-Z]*[pP])assword=\"[^\"]*\"", "$1assword=\"********\"");
-        // property: default-repository-config
-        xml = xml.replaceAll("([a-zA-Z]*[pP])assword\">[^<]*<", "$1assword\">********<");
-        return xml;
+        String res = xml;
+        List<String> keywords = getKeywordList(SECURE_KEYWORDS_PROPERTY, DEFAULT_SECURE_KEYWORDS);
+        List<String> whitelist = getKeywordList(WHITELISTED_KEYWORDS_PROPERTY, DEFAULT_WHITELISTED_KEYWORDS);
+        for (String kw : keywords) {
+            if (res.contains(kw)) {
+                for (String pattern : List.of(
+                        // node startswith
+                        String.format("(?<start><(?<key>\\w*%s)\\s*>)[^<]*(?<end></\\w*%s>)", kw, kw),
+                        // node endswith
+                        String.format("(?<start><(?<key>%s\\w*)\\s*>)[^<]*(?<end></%s\\w*>)", kw, kw),
+                        // attributes startswith
+                        String.format("(?<start>(?<key>\\w*%s)=\")[^\"]*(?<end>\")", kw),
+                        String.format("(?<start>(?<key>\\w*%s)\"\\s*>)[^<]*(?<end><)", kw),
+                        // attributes endswith
+                        String.format("(?<start>(?<key>%s\\w*)=\")[^\"]*(?<end>\")", kw),
+                        String.format("(?<start>(?<key>%s\\w*)\"\\s*>)[^<]*(?<end><)", kw))) {
+                    res = secureXML(res, pattern, whitelist);
+                }
+            }
+        }
+        return res;
+    }
+
+    protected static String secureXML(String xml, String pattern, List<String> whitelist) {
+        StringBuffer result = new StringBuffer();
+        Matcher m = Pattern.compile(pattern).matcher(xml);
+        while (m.find()) {
+            String replacement;
+            if (whitelist.contains(m.group("key"))) {
+                replacement = m.group();
+            } else {
+                replacement = m.group("start") + SECRET_VALUE + m.group("end");
+            }
+            m.appendReplacement(result, replacement);
+        }
+        return m.appendTail(result).toString();
+    }
+
+    protected static List<String> getKeywordList(String property, List<String> defaultValue) {
+        ConfigurationService cs = Framework.getService(ConfigurationService.class);
+        String value = cs.getString(property, null);
+        if (value == null) {
+            return defaultValue;
+        }
+        return List.of(value.split("\\s*,[,\\s]*"));
```

Comment:
Something like below should be used as the ConfigurationService APIs with default value are not designed to pass null.
```
return cs.getString(property).map(v -> v.split("\\s*,[,\\s]*")).map(List::of).orElse(defaultValue);
```